Client/Frontend/Shields/ShieldsViewController.swift

```diff
@@ -8,41 +8,45 @@ import Shared
 import BraveShared
 import Data
 
-class ShieldBlockedStats: NSObject {
-    var adsAndTrackers = 0
-    var httpsUpgrades = 0
-    var blockedScripts = 0
-    var fingerprintingProtection = 0
-}
-
 /// Displays shield settings and shield stats for a given URL
 class ShieldsViewController: UIViewController, PopoverContentComponent {
+    let tab: Tab
+    
     /// The url loaded currently. Update this as the pages url changes
-    var url: URL? {
+    private var url: URL? {
         didSet {
             updateToggleStatus()
         }
     }
     /// The blocked stats. Update this as the pages block stats change
-    var shieldBlockStats: ShieldBlockedStats {
+    private var shieldBlockStats: TPPageStats {
         didSet {
             updateShieldBlockStats()
         }
     }
     
     var shieldsSettingsChanged: ((ShieldsViewController) -> Void)?
     
+    private var statsUpdateObservable: AnyObject?
+    
     /// Create with an initial URL and block stats (or nil if you are not on any web page)
-    init(url: URL?, shieldBlockStats: ShieldBlockedStats) {
-        self.url = url
-        self.shieldBlockStats = shieldBlockStats
+    init(tab: Tab) {
+        self.tab = tab
+        self.url = tab.url
+        self.shieldBlockStats = tab.contentBlocker.stats
```

Comment:
If we are retaining `tab`, is there any point in storing these as additional properties? It also then requires keeping it updated, as in [here](https://github.com/brave/brave-ios/pull/403/files#diff-ed29423ac71a29faa319243941d34b93R48)